wordpress-mobile/WordPress-FluxC-Android
# Issue/7857 generate product variants

Closes: https://github.com/woocommerce/woocommerce-android/issues/7857

### Why
We need to be able to create many variations for the Generate all combinations of variations project.
Creating all the variations in one request is the solution we agreed on after discussion, taking into account that it is fast and makes good use of resources. To achieve this solution, we'll use this endpoint from Woo REST API: https://woocommerce.github.io/woocommerce-rest-api-docs/#batch-update-product-variations.

### Description
This PR adds support to create product variations using the batch-update-product-variations endpoint. It accomplishes its goal by modifying the batchUpdateVariations function and adding the missing parameters to match the API request. It also adds a UI to the FluxC example app to make it easier to test the request.

### Testing instructions

1. Open the FluxC example app
2. Tap on Woo
3. Tap on Select Site
4. Select a Site
5. Tap on Products
6. Tap on Select Site
7. Select a Site
8. Scroll to Batch Generate Variations 
9. Tap on Batch Generate Variations 
10. Enter a valid variable product id
11. Tap on Set Product ID
12. Enter a valid Attribute Id, Name, and Option
13. Tap on Add Attribute
14. Tap on Add Variation
15. Repeat steps 12 to 14 (as many variations as you want to add)
16. Tap on Generate Variation
17. Check that the variations are created successfully

### Images

| Adding variants | Success response |
| ------------- | ------------- |
| <img width="300" src="https://user-images.githubusercontent.com/18119390/202299981-b134620f-ba28-4b9a-ad46-264059fe1e5a.png" />  | <img width="300" src="https://user-images.githubusercontent.com/18119390/202299987-6aba77c4-c417-4b6d-b6b4-e778721e2fc1.png" />  |

![zCfsHr.png](https://user-images.githubusercontent.com/18119390/202300704-43d07a69-076c-4b73-ac68-7b1ae361c3db.png)
